install/0000_30_machine-api-operator_07_machinehealthcheck.crd.yaml

```diff
@@ -32,136 +20,150 @@ spec:
     singular: machinehealthcheck
   scope: Namespaced
   preserveUnknownFields: false
```

Comment:
Can we drop `preserveUnknownFields:  false`? it's default in `apiextensions.k8s.io/v1`. "This can either be true or undefined. False is forbidden." https://book.kubebuilder.io/reference/markers/crd-processing.html

Reply:
I believe that is for the values within the OpenAPIV3Schema block. 

As I understand it, this value is required to enable upgrades to work smoothly

>For CustomResourceDefinitions created in apiextensions.k8s.io/v1, structural OpenAPI v3 validation schemas are required and pruning is enabled and cannot be disabled (note that CRDs converted from apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1 might lack structural schemas, and spec.preserveUnknownFields might be true).

From: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#pruning-versus-preserving-unknown-fields

Because the definitions have been created using the `extensions/v1beta1` API up until now and didn't have structural schemas, the `spec.PreserveUnknownFields` value has been set to true, even when stored as `extensions/v1`. 
So we need to force the value to `false` to ensure the CRD starts being pruned and restores the behaviour to the default behaviour for `v1`

Reply:
cc @sttts to confirm

Reply:
Posted a message to sig-api-machinery (https://kubernetes.slack.com/archives/C0EG7JC6T/p1586869328387600) and got a confirmation there that the behaviour is as I've described

Reply:
false is the default. If I see this correctly it only depends only on your apply mechanism. These CRDs are applied through CVO?

Reply:
Correct, these are applied by CVO, do we not need the field then? It was my understanding that if the CRD was created as v1beta1, it would be stuck with `preserveUnknownFields: true` if read/updated as v1, unless explicitly overwritten, is that not the case?

Reply:
As I wrote: it depends on CVO's apply mechanism. If it reads your former v1beta1 CRD as v1, it won't know that it was originally a v1beta1 CRD. I assume it will compare the v1 in cluster to the v1 from your manifests. Read the CVO code to find out whether it takes `preserveUnknownFields` into consideration.

Reply:
Ok, having a look through, it seems that applying a CRD via CVO "stomps" the spec, and then sends an update rather than doing a merge as I had expected (like kubectl apply), so I think we can drop preserveUnknownFields, thanks @sttts 

https://github.com/openshift/cluster-version-operator/blob/ee4cd7ea187de8f65eed1a57937254fbe8810308/lib/resourcemerge/apiext.go#L61-L65

https://github.com/openshift/cluster-version-operator/blob/ee4cd7ea187de8f65eed1a57937254fbe8810308/lib/resourceapply/apiext.go#L56-L63